forge/db/controllers/ProjectTemplate.js

```diff
@@ -198,6 +198,16 @@ module.exports = {
         if (typeof result.httpNodeAuth?.pass === 'string' && result.httpNodeAuth.pass.length > 0) {
             result.httpNodeAuth.pass = hash(result.httpNodeAuth.pass)
         }
+        if (result.apiMaxLength) {
+            if (!/^[0-9]+[km]?$/.test(result.apiMaxLength)) {
+                throw new Error('Invalid settings.apiMaxLength')
+            }
+        }
+        if (result.debugMaxLenth) {
+            if (!Number.isInteger(result.debugMaxLength)) {
+                throw new Error('Invalid settings.debugMaxLength')
+            }
```

Comment:
```suggestion
            if (!Number.isInteger(result.debugMaxLength) || result.debugMaxLength < 0) {
                throw new Error('Invalid settings.debugMaxLength')
            }
```

Should prevent invalid negative numbers at API level

Reply:
Should we not consider an upper limit? greater than `10000` is pushing it but allowing unlimited is possibly a good thing

Reply:
This is a performance thing, if a user cripples themselves it's on them.